### PR TITLE
Support to use SQS & SNS endpoints in session

### PR DESCRIPTION
**Pull request by jkraj**
Presently SQS & SNS endpoints are not considered during session
creation. When deploying pubsub along with aws localstack it is
failing because it is using default endpoint to create sessions.

### CODE REVIEW

**Comment by jkraj** [comment]
> Could you explain your application and when you would pass a aws local stack?

I am using [https://github.com/Infoblox-CTO/saas-app-deployment/tree/master/support-tools/minikube](https://github.com/Infoblox-CTO/saas-app-deployment/tree/master/support-tools/minikube) setup to validated atlas onprem control-plane components but presently it just verify the deployment of cloud components. CDC apps depends on Configmanager which needs pubsub. Idea here to point that pubsub to localstack. 

> Overall looks good. Could you add a check to verify the credentials.
Because we use dummy AWS_token and AWS_key credentials verification is not needed.

**Comment by razamiDev** [comment]
@jkraj could you fix the codecov/patch failure.

File: examples/server/main.go [hunk 1]
```diff
@@ -98,7 +112,37 @@ func newAWSPubSubServer(logger *logrus.Logger) (pubsubgrpc.PubSubServer, error)
...
 	subFactory := func(ctx context.Context, topic, subscriptionID string) (pubsub.Subscriber, error) {
-		return pubsubaws.NewSubscriber(sess, topic, subscriptionID)
+		return pubsubaws.NewSubscriber(sess, sess, topic, subscriptionID)
+	}
+	return pubsubgrpc.NewPubSubServer(pubFactory, subFactory), nil
+}
+
+// newLocalStackPubSubServer creates a new grpc PubSub server using the broker
+// implementation for LocalStack
+func newLocalStackPubSubServer(logger *logrus.Logger) (pubsubgrpc.PubSubServer, error) {
```
Comment (chinmayb) [inline examples/server/main.go]: I feel we dont need a separate method for localstack, instead we could create two sessions for sns and sqs by reading endpoints from conf. and change the signature of `VerifyPermissions`

**Comment by Evgeniy-L** [comment]
closing as there was no actions for a long time

**Pull request closed without merging**